executor/executor.go

```diff
@@ -88,14 +88,22 @@ func NewMesosExecutorDriver(config DriverConfig) (*MesosExecutorDriver, error) {
 	}
 
 	driver := &MesosExecutorDriver{
-		exec:      config.Executor,
-		status:    mesosproto.Status_DRIVER_NOT_STARTED,
-		stopCh:    make(chan struct{}),
-		destroyCh: make(chan struct{}),
-		stopped:   true,
-		updates:   make(map[string]*mesosproto.StatusUpdate),
-		tasks:     make(map[string]*mesosproto.TaskInfo),
-		workDir:   ".",
+		status:  mesosproto.Status_DRIVER_NOT_STARTED,
+		stopCh:  make(chan struct{}),
+		updates: make(map[string]*mesosproto.StatusUpdate),
+		tasks:   make(map[string]*mesosproto.TaskInfo),
+		workDir: ".",
+		started: make(chan struct{}),
+	}
+	driver.cond = sync.NewCond(&driver.lock)
+	// decouple serialized executor callback execution from goroutines of this driver
+	var execLock sync.Mutex
+	driver.withExecutor = func(f func(e Executor)) {
+		go func() {
```

Comment:
status updates are ack'd -- the scheduler won't get another status update for the same task until a prior update message has been ack'd. this means that we should do our best to preserve the order of execution w/ respect to (at least) status update messages.